store/src/java/com/zimbra/cs/ephemeral/EphemeralStore.java

```diff
@@ -186,27 +189,36 @@ public static final void setFactory(Class<? extends Factory> factoryClass) {
     }
 
     public static final void setFactory(Class<? extends Factory> factoryClass, FailureMode onFailure) {
+        String className = factoryClass.getDeclaringClass().getSimpleName();
         try {
-            boolean useFallback = false;
+            boolean useForwarding = false;
             Factory fac = factoryClass.newInstance();
-            if (!(fac instanceof LdapEphemeralStore.Factory)) {
-                EphemeralStore store = fac.getStore();
-                MigrationFlag flag = AttributeMigration.getMigrationFlag(store);
+            try {
+                MigrationInfo info = AttributeMigration.getMigrationInfo();
+                Status curStatus = info.getStatus();
+                if (!Strings.isNullOrEmpty(info.getURL()) && (curStatus == Status.COMPLETED || curStatus == Status.IN_PROGRESS)) {
```

Comment:
If the migration failed, the assumption is that it will have to be rerun. In that case, I don't think forwarding is not necessary.